src/lib/user-data-watcher.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import { getMainWindow } from 'src/main-window';
+import { getUserDataFilePath } from 'src/storage/paths';
+import { loadUserData } from 'src/storage/user-data';
+
+let watcher: fs.FSWatcher | null = null;
+
+export function startUserDataWatcher() {
+	if ( watcher ) {
+		return;
+	}
+
+	const filePath = getUserDataFilePath();
+
+	const fsEventHandler = ( eventType: string ) => {
+		loadAndUpdateUserData();
+		if ( eventType === 'rename' && watcher ) {
+			watcher.close();
+			watcher = fs.watch( filePath, fsEventHandler );
+		}
+	};
+
+	watcher = fs.watch( filePath, fsEventHandler );
+
+	getMainWindow().then( ( mainWindow ) => {
+		if ( mainWindow.webContents.isLoading() ) {
+			mainWindow.webContents.once( 'did-finish-load', () => {
+				loadAndUpdateUserData();
+			} );
+		} else {
+			loadAndUpdateUserData();
+		}
+	} );
```

Comment:
```suggestion
	const mainWindow = await getMainWindow();

	if ( mainWindow.webContents.isLoading() ) {
		mainWindow.webContents.once( 'did-finish-load', () => {
			loadAndUpdateUserData();
		} );
	} else {
		loadAndUpdateUserData();
	}
```
Nit, but I would make the function async and `await` the result of `getMainWindow`